### ISSUE TITLE
yarn versions and upgrading

### ISSUE THREAD

**Issue by gilzow** [platformsh/source-operations#29]
Later versions of yarn (after 2.x maybe?) have replaced the `upgrade` command with `up` : https://yarnpkg.com/cli/up
Later versions also suggest installing yarn via corepack https://yarnpkg.com/getting-started/install 
And it appears you can still use yarn via corepack with `corepack yarn` even if you haven't configured yarn as the package management system you want to use. 

We can't run `corepack enable` and have it add symlinks to the package management binaries because that part of the OS is read-only, even during build:

```
> corepack enable
    Internal Error: EROFS: read-only file system, symlink '../lib/node_modules/corepack/dist/pnpm.js' -> '/usr/bin/pnpm'
    Error: EROFS: read-only file system, symlink '../lib/node_modules/corepack/dist/pnpm.js' -> '/usr/bin/pnpm'
```

This makes it particularly challenging to know which way to use yarn (`yarn` vs `corepack yarn`) and then which command we need to issue in order to run the package updates (`yarn upgrade` vs `yarn up`) as we'd need to test which version of yarn is being used before we know which command to issue. 

It does appear that corepack will default to `v1.22.19` if no specific version for yarn has been given. However, we're going to need to think of a solution once we have a template that is using a later version of yarn.

**Comment by gilzow** [comment]
also, despite the yarn docs stating you should install yarn via corepack, according to the [node docs](https://nodejs.org/api/corepack.html), corepack is experimental. 

>[Stability: 1](https://nodejs.org/api/documentation.html#stability-index) - Experimental
[Corepack](https://github.com/nodejs/corepack) is an experimental tool to help with managing versions of your package managers

**Comment by gilzow** [comment]
closed via #28
